### ISSUE TITLE
fix: container not being restarted even with "unless-stopped" mode

### ISSUE THREAD

**Issue by ntorga** [goinfinite/ez-issue-tracker#84]
During the demo deployment, I realized the container is being killed for some reason:
```
time="2024-06-20T04:29:42Z" level=info msg="receive a signal to stop all process & exit" signal=terminated
```

However, podman is suppose to restart the container due to the "unless-stopped":
```
$ demo@control:~> podman inspect e7d3392f22f4
               "RestartPolicy": {
                    "Name": "unless-stopped",
                    "MaximumRetryCount": 0
               },
```

This is likely a podman/microOS issue, however, we need to investigate and maybe create a fallback routine so if podman fails to restart, Control comes to the rescue.

**Comment by ntorga** [comment]
Ok, checking the instance logs, I found this:
```
Jun 20 04:29:42 control.demo.speedia.net rebootmgrd[821]: rebootmgr: reboot triggered now!
Jun 20 04:29:42 control.demo.speedia.net systemd-logind[786]: The system will reboot now!
Jun 20 04:29:42 control.demo.speedia.net systemd-logind[786]: System is rebooting.
```

There is some info on the documentation regarding it:
https://en.opensuse.org/Kubic:Update_and_Reboot

The instance was rebooted after a kernel update. There is a maintenance window configured by default:
```
control:~ # rebootmgrctl get-window
Maintenance window is set to *-*-* 03:30:00, lasting 01h30m.

control:~ # systemctl status transactional-update.timer
● transactional-update.timer - Daily update of the system
     Loaded: loaded (/usr/lib/systemd/system/transactional-update.timer; enabled; preset: enabled)
     Active: active (waiting) since Thu 2024-06-20 04:31:27 UTC; 12h ago
    Trigger: Fri 2024-06-21 00:55:43 UTC; 7h left
   Triggers: ● transactional-update.service
       Docs: man:transactional-update(8)

Jun 20 04:31:27 control.demo.speedia.net systemd[1]: Started Daily update of the system.
```

A daily reboot will likely prevent backups from completing and may crash some database. Once a week should be ok:
```
mkdir -p /etc/systemd/system/transactional-update.timer.d/
cat >"/etc/systemd/system/transactional-update.timer.d/timer.conf" <<EOF
[Timer]
OnCalendar=
OnCalendar=Sat *-*-* 06:00:00
EOF
systemctl daemon-reload
systemctl list-timers | grep 'update'
```

I added this to the installation code but with a random generator so "06" is replace with an hour between 0 and 6 to avoid network overload.

The problem is that podman is suppose to reboot all containers once the instance is up but that didn't happen.

Control does have a boot middleware to power on all the containers on the wrong state, but that didn't happen either.

Looking at the code I can see the problem. Since the container isn't stopped by Control, the database holds that the container is still up, thus it won't change the status of the container.

I could add a power off and power on into the boot logic, but that would power off containers that podman did manage to start. Instead, it would be beneficial to have a routine that runs on the start and every now and then to update the status of the containers according to podman cli. That way if a container stops and can't be restarted due to its restart policy, the failed state would reflect on the database. See goinfinite/ez-issue-tracker#83.

Anyhow, moving now to investigate the podman issue of not restarting the container.

**Comment by ntorga** [comment]
What a rabbit hole. I had no ideia it would take me almost a day and half to fix this.

The systemd unit generated by podman was deprecated in favor of Quadlet:
https://www.redhat.com/sysadmin/quadlet-podman

After learning and playing with Quadlet, I noticed that the generator used there adds the flags --rm and --replace to all containers, so it's not made for stateful containers (what is, am I right?):
https://github.com/containers/podman/blob/main/pkg/systemd/quadlet/quadlet.go

Instead of bothering the good folks of podman team and waiting, I decided that Control would manage the system unit by itself so I refactored the deployment code in a way that the systemd unit file is created as the unprivileged user with all the nice configs, including resources constrains.

I also had to add a PreStart so that Podman still updates the resource constrains during runtime cause if systemd is reloaded with updated cpu/memory quotas, podman does not pick up and won't update the cgroups files inside the container (cpu.max and memory.max).

The live version now has those fixes, although they are breaking changes. What isn't in a alpha state tho? Oh well. The demo server was updated to reflect these changes and after the reboot and killing the supervisord process on the container, systemd managed to restart the container smoothly.
